### PR TITLE
add .npmignore to limit unnecessary files in package

### PR DESCRIPTION
**Pull request by kellyselden**

### CODE REVIEW

File: .npmignore [hunk 1]
```diff
@@ -0,0 +1,9 @@
+assets
+blueprints/.jshintrc
+dev
+docs
+tests
```
Comment (stefanpenner) [inline .npmignore]: please depend on some of the test helpers

Comment (kellyselden) [inline .npmignore]: fixed

**Comment by stefanpenner** [comment]
I think this is good, @rwjblue can you :eyes: you are likely to remember some of the usages.

**Comment by kellyselden** [comment]
I need to make some changes, hold off on merge please

**Comment by kellyselden** [comment]
made changes, gotta keep the beginning slash to only check root folders. also wildcard exclude all files at root not in folders because the package doesn't need any

https://docs.npmjs.com/misc/developers#keeping-files-out-of-your-package

here's the result of `npm pack`

capture

@stefanpenner @rwjblue ready for review again.

**Comment by kellyselden** [comment]
friendly ping @stefanpenner @rwjblue

File: .npmignore [hunk 2]
```diff
@@ -0,0 +1,7 @@
+/assets
+/blueprints/.jshintrc
+/dev
+/docs
+/tests/!(helpers)
+/tmp
+/*.*
```
Comment (stefanpenner) [inline .npmignore]: what does this match?

Comment (kellyselden) [inline .npmignore]: all root level files, aside from the automatically excluded readme, changlog, license, and package.json

Comment (kellyselden) [inline .npmignore]: @stefanpenner did you see the image here? https://github.com/ember-cli/ember-cli/pull/5356#issuecomment-172391663

Comment (stefanpenner) [inline .npmignore]: Ah so this doesn't include package.json right? That one is essentially white-listed.

Comment (rwjblue) [inline .npmignore]: Confirm. There are a list of files that you cannot ignore (per npm).

**Pull request merged**